src/Symfony/Component/ExpressionLanguage/Node/BinaryNode.php

```diff
@@ -154,4 +154,9 @@ public function evaluate($functions, $values)
                 return preg_match($right, $left);
         }
     }
+
+    public function dump()
+    {
+        return sprintf('%s %s %s', $this->nodes['left']->dump(), $this->attributes['operator'], $this->nodes['right']->dump());
```

Comment:
you should add braces, so that ``a + (b * c)`` and ``(a + b) * c`` are not dumped the same way. A fully-braced expression is unambiguous (while verbose) while dumping without braces simply cannot represent everything (when using an infix notation)